schema.json

```diff
@@ -1,255 +1,266 @@
-              "type": "integer",
-              "minimum": 0,
-              "exclusiveMinimum": true
-            },
-            "min_size": {
-              "type": "integer",
-              "minimum": 0
-            }
-          },
-          "required": ["name", "fields", "type"]
+        "unit": {
+          "description": "The unit type, defined in Table A-13 of E1.20.",
+          "type": "integer",
+          "minimum": 0,
+          "maximum": 255
         }
-      ]
+      },
+      "required": [ "type" ]
+    },
+    "labeledIntegerType": {
+      "description": "Associates a name to an integer value.",
+      "type": "object",
+      "properties": {
+        "name": { "$ref": "#/definitions/nameType" },
+        "value": { "type": "integer" }
+      },
+      "required": [ "value" ]
     },
-    "label": {
+    "listType": {
+      "description": "A list of objects all having the same type.",
       "type": "object",
       "properties": {
-        "label": {
-          "minLength": 1,
-          "type": "string"
+        "name": { "$ref": "#/definitions/nameType" },
+        "type": { "const": "list" },
+        "itemType": {
+          "$ref": "#/definitions/oneOfTypes"
         },
-        "value": {
+        "minItems": {
+          "description": "The minimum list size.",
+          "type": "integer",
+          "minimum": 0
+        },
+        "maxItems": {
+          "description": "The maximum list size.",
           "type": "integer",
-          "maximum": 4294967295,
           "minimum": 0
         }
       },
-      "required": ["label", "value"]
+      "required": [ "type", "itemType" ]
     },
-    "name": {
+    "nameType": {
+      "description": "A name is a string having a length of at least 1.",
       "type": "string",
       "minLength": 1
     },
-    "range": {
+    "oneOfTypes": {
+      "description": "One of any of the types. This provides a single location
+                      to keep the list.",
+      "oneOf": [
+        { "$ref": "#/definitions/bitFieldType" },
+        { "$ref": "#/definitions/compoundType" },
+        { "$ref": "#/definitions/integerType" },
+        { "$ref": "#/definitions/listType" },
+        { "$ref": "#/definitions/pdEnvelopeType" },
+        { "$ref": "#/definitions/simpleType" },
+        { "$ref": "#/definitions/stringType" },
+        { "$ref": "#/definitions/subdeviceType" }
+      ]
+    },
+    "subdeviceType": {
+      "description": "A subdevice type. One of the root device (e.g. value 0),
+                      the root or all subdevices or the broadcast subdevice,
+                      the root or any subdevice, or just any subdevice.",
+      "type" : "string",
+      "enum": [
+            "root",
+            "root_or_any_subdevice_or_broadcast",
+            "root_or_any_subdevice",
+            "any_subdevice"
+          ]
+    },
+    "pdEnvelopeType": {
+      "description": "Contains a length/data pair for one Parameter Data item,
```

Comment:
I'm trying to work out if the data needs to be included too here, but I'm not sure if it does at the level of the schema we're in, I think that would be down to the next level down.

Reply:
Something of type "pdEnvelopeType" is understood to contain "length" bytes of data. In the schema, it actually represents the data, just like the "int64" type represents 8 bytes of data.